File: 1029-two-city-scheduling/1029-two-city-scheduling.py
class Solution:
    def twoCitySchedCost(self, costs: List[List[int]]) -> int:

        l = sorted(costs, key = lambda x : x[0]-x[1])
        n = len(costs)//2
        total = 0
        for i in range(n):
            total += l[i][0]+l[i+n][1]
        return total

File: 1029-two-city-scheduling/1029-two-city-scheduling.cpp
class Solution {
public:
    int twoCitySchedCost(vector<vector<int>>& costs) 
    {
        sort(costs.begin(),costs.end(),[](auto &x, auto &y){return x[0]-x[1]<y[0]-y[1];});
        int n = costs.size();
        int n_A = n/2;
        int sum = 0;
        for(int i = 0; i<n;++i)
        {
            if(i<n_A)
            {
                sum += costs[i][0];
            }
            else
            {
                sum += costs[i][1];
            }
        }
        return sum;
        
    }
};